src/cpp/_pythia8.cpp

```diff
@@ -103,6 +103,32 @@ py::array_t<int> event_array_children(Event &event)
     return py::array_t<int>(shape, strides, ptr1);
 }
 
+// refills "event" stack with particles
+void fill(Event &event,
+                    py::array_t<int> &pid,
+                    py::array_t<int> &status,
+                    py::array_t<double> &px,
+                    py::array_t<double> &py,
+                    py::array_t<double> &pz,
+                    py::array_t<double> &energy,
+                    py::array_t<double> &mass) 
+{
+    // Get a raw reference to numpy array
+    auto pid_ = pid.unchecked<1>();
+    auto status_ = status.unchecked<1>();
+    auto px_ = px.unchecked<1>();
+    auto py_ = py.unchecked<1>();
+    auto pz_ = pz.unchecked<1>();
+    auto energy_ = energy.unchecked<1>();
+    auto mass_ = mass.unchecked<1>();
+
+    event.reset();
+    for (int i = 0; i != pid.size(); ++i) {
+        event.append(pid_[i], status_[i], 0, 0,
+                      px_[i], py_[i], pz_[i], energy_[i], mass_[i]);
+    }    
+}
```

Comment:
Please adjust the indentation or install an autoformatter. We should add one to pre-commit...

Reply:
Adjusted